### ISSUE TITLE
Validate on Pull request

### ISSUE THREAD

**Issue by sandeep1989** [forcedotcom/devops-center-feedback#234]
**Is your feature request related to a problem? Please describe.**
Validating the changes being submitted through a pull request will help identify any conflicts or test failures
**Describe the solution you'd like**
Run a validate on a PR and provide the feedback to the dev to review. And make the "ready to promote" be vailable only if validation succeeds.
**Additional context**
Current workaround is to use Github actions to run the validation on creation of a PR.

**Comment by GilsonCanario** [comment]
Hi @sandeep1989, thank you for your feature request. Could you help us understand what is your definition of "validating a PR"? Are you talking about Validate-Only deployment, static analysis or something else entirely?

**Comment by sandeep1989** [comment]
Yes Validate Only deployment - to validate the changes being submitted by the dev and make sure the work item do not have any missing dependencies or test class failures before the Work item is ready to be reviewed.
